SQL/Archive/2018-05-28_publication_schema.sql

```diff
@@ -0,0 +1,159 @@
+SET FOREIGN_KEY_CHECKS=0;
+-- Publication Status
+DROP TABLE IF EXISTS publication_status;
+CREATE TABLE `publication_status` (
+  `PublicationStatusID` int(2) unsigned NOT NULL,
+  `Label` varchar(255) NOT NULL,
+  PRIMARY KEY(`PublicationStatusID`)
+) ENGINE=InnoDB DEFAULT CHARSET='utf8';
+DELETE FROM publication_status;
+INSERT INTO publication_status VALUES (1, 'Pending');
+INSERT INTO publication_status VALUES (2, 'Approved');
+INSERT INTO publication_status VALUES (3, 'Rejected');
+
+-- Main table
+DROP TABLE IF EXISTS publication;
+CREATE TABLE `publication` (
+    `PublicationID` int(10) unsigned NOT NULL AUTO_INCREMENT,
+    `PublicationStatusID` int(2) unsigned NOT NULL default 1,
+    `UserID` int(10) unsigned NOT NULL,
+    `RatedBy` int(10) unsigned,
+    `DateProposed` date NOT NULL,
+    `DateRated` date default NULL,
+    `Title` varchar(255) NOT NULL,
+    `LeadInvestigator` varchar(255) NOT NULL,
+    `LeadInvestigatorEmail` varchar(255) NOT NULL,
+    `RejectedReason` varchar(255) default NULL,
+    `Description` text NOT NULL,
+    PRIMARY KEY(`PublicationID`),
+    CONSTRAINT `FK_publication_1` FOREIGN KEY(`UserID`) REFERENCES `users` (`ID`),
+    CONSTRAINT `FK_publication_2` FOREIGN KEY(`RatedBy`) REFERENCES `users` (`ID`),
+    CONSTRAINT `FK_publication_3` FOREIGN KEY(`PublicationStatusID`) REFERENCES `publication_status` (`PublicationStatusID`)
+) ENGINE=InnoDB DEFAULT CHARSET='utf8';
+
+-- Separate table for Keywords
+DROP TABLE IF EXISTS publication_keyword;
+CREATE TABLE `publication_keyword` (
+  `PublicationKeywordID` int(10) unsigned NOT NULL AUTO_INCREMENT,
+  `Label` varchar(255) NOT NULL,
+  PRIMARY KEY(`PublicationKeywordID`)
+) ENGINE=InnoDB DEFAULT CHARSET='utf8';
+
+-- Publication - Keyword relational table
+DROP TABLE IF EXISTS publication_keyword_rel;
+CREATE TABLE `publication_keyword_rel` (
+  `PublicationID` int(10) unsigned NOT NULL,
+  `PublicationKeywordID` int(10) unsigned NOT NULL,
+  CONSTRAINT `UC_publication_keyword_1` UNIQUE (PublicationID, PublicationKeywordID),
+  CONSTRAINT `FK_publication_keyword_1` FOREIGN KEY(`PublicationID`) REFERENCES `publication` (`PublicationID`),
+  CONSTRAINT `FK_publication_keyword_2` FOREIGN KEY(`PublicationKeywordID`) REFERENCES `publication_keyword` (`PublicationKeywordID`)
+) ENGINE=InnoDB DEFAULT CHARSET='utf8';
+
+
+DROP TABLE IF EXISTS publication_collaborator;
+CREATE TABLE `publication_collaborator` (
+  `PublicationCollaboratorID` int(10) unsigned NOT NULL AUTO_INCREMENT,
+  `Name` varchar(255) NOT NULL,
+  `Email` varchar(255),
+  PRIMARY KEY(`PublicationCollaboratorID`),
+  UNIQUE (`Name`,`Email`)
+) ENGINE=InnoDB DEFAULT CHARSET='utf8';
```

Comment:
This would create a uniqueness on both attributes combined! 
2 person with different name can have the same email.
If you want uniqueness on Name, then create 2 unique constraints